### PR TITLE
Fixes issue 1077, removes separate args for mkl and scipy

### PR DESCRIPTION
**Pull request by tehruhn**
Unified the args list for scipy and mkl solvers, and fixed issue 1077

### CODE REVIEW

**Comment by tehruhn** [comment]
So Travis CI says there's an error in smesolve, which I haven't touched. :/

**Comment by nathanshammah** [comment]
There is a bunch of .xml and .iml files that keep being tracked in your PR, I expect this is not part of the refactoring but just due to your local environment? You can fix it setting up a specific gitignore in your workspace I think.

**Comment by nathanshammah** [comment]
I am referencing #1077 explicitly here.

**Comment by nathanshammah** [comment]
This addresses also #1017 if I understand correctly. Looks good to me but would prefer to know @Ericgig's opinion on this.

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,7 @@
 *.so
 .DS_Store
 .f2py_f2cmap
+.idea/
```
Comment (Ericgig) [inline .gitignore]: Why is this needed?

File: qutip/steadystate.py [hunk 2]
```diff
@@ -255,12 +243,13 @@ def steadystate(A, c_op_list=[], method='direct', solver=None, **kwargs):
...
-    elif solver == 'mkl':
-        ss_args = _mkl_steadystate_args()
-    else:
-        raise Exception('Invalid solver keyword argument.')
+    # if solver == 'scipy':
+    #     ss_args = _default_steadystate_args()
+    # elif solver == 'mkl':
+    #     ss_args = _mkl_steadystate_args()
+    # else:
+    #     raise Exception('Invalid solver keyword argument.')
```
Comment (Ericgig) [inline qutip/steadystate.py]: Remove these comments before merging.

File: qutip/steadystate.py [hunk 3]
```diff
@@ -255,12 +243,13 @@ def steadystate(A, c_op_list=[], method='direct', solver=None, **kwargs):
...
+    ss_args = _default_steadystate_args()
     ss_args['method'] = method
```
Comment (Ericgig) [inline qutip/steadystate.py]: The `solver` value need to be set to `mkl` or `scipy` as needed. It is always `scipy` now.

File: .gitignore (hunk 1 above)
Comment (tehruhn) [reply .gitignore]: Because I use PyCharm for its debugger, and PyCharm automatically creates the .idea/ folder in my workspace. I thought I'd add it there in case someone else did the same, and to avoid having to manually not commit the folder in my workspace.

File: qutip/steadystate.py (hunk 2 above)
Comment (tehruhn) [reply qutip/steadystate.py]: Done in latest commit.

File: qutip/steadystate.py (hunk 3 above)
Comment (tehruhn) [reply qutip/steadystate.py]: I assumed that it would be up to the user to specify if scipy or mkl was needed? I set scipy as default because I thought it was the most stable solver. Could you tell me what specifically I should set the default as, or what changes can be made?

Comment (Ericgig) [reply qutip/steadystate.py]: Yes but it is in the variable `solver`, not in `kwargs` so it will not be updated in `ss_args`. Also it should be done before the line `ss_args['info']['solver'] = ss_args['solver']`. Or the info will always says `sicpy` even if mkl was used.

Comment (tehruhn) [reply qutip/steadystate.py]: Aah, I see the issue. Apologies for the oversight.

Comment (tehruhn) [reply qutip/steadystate.py]: I've fixed it in the latest commit, please check again. And thanks!
